electron-init.ts

```diff
@@ -127,21 +41,108 @@ function init() {
         skybinHome = process.env.SKYBIN_HOME;
     }
 
-    console.log('looking for skybin home folder at', skybinHome);
+    isRenterSetup = fs.existsSync(`${skybinHome}/renter`);
+    isProviderSetup = fs.existsSync(`${skybinHome}/provider`);
 
-    isSkybinSetup = fs.existsSync(skybinHome);
-    if (isSkybinSetup) {
-        console.log('skybin homefolder found');
-
-        // TODO: check that services are running. If not, find out why, attempt to run them, etc.
-
-    } else {
-        console.log('no skybin homefolder found');
+    if (isRenterSetup) {
+        renterConfig = loadConfig(`${skybinHome}/renter/config.json`);
+    }
+    if (isProviderSetup) {
+        providerConfig = loadConfig(`${skybinHome}/provider/config.json`);
     }
 
     launchApp();
 }
 
+ipcMain
+    .on('isRenterSetup', (event) => {
+        event.returnValue = isRenterSetup;
+    })
+    .on('setupRenter', (event, options) => {
+        try {
+            initRenter(options);
+            initRenterDaemon();
+        } catch (err) {
+            event.sender.send('setupRenterDone', {
+                error: err.toString()
+            });
+            return;
+        }
+
+        isRenterSetup = true;
+        try {
+            renterConfig = loadConfig(`${skybinHome}/renter/config.json`);
+        } catch (err) {
+            event.sender.send('setupRenterDone', {
+                error: `Unable to load skybin renter config. Error: ${err}`
+            });
+            return;
+        }
+
+        // Wait 1 second and ensure we can ping the renter.
+        setTimeout(() => {
```

Comment:
Async wait for an arbitrary amount of time seems like it could be subject to the odd case where the renter isn't scheduled (and run) by the CPU within the time frame and causes issues. I saw problems like this when I tried to hack together this functionality.